packages/manager/modules/bm-server-components/src/ipmi/controller.js

```diff
@@ -0,0 +1,690 @@
+      testIpmi: false,
+      otherTask: false,
+      localTask: false,
+    };
+
+    this.ssh = {
+      list: [],
+      error: false,
+      selectedKey: '',
+    };
+
+    this.ipmiHelpUrl = getIpmiGuideUrl(this.user.ovhSubsidiary);
+
+    this.loader.loading = true;
+    this.loader.error = false;
+    this.$q
+      .all({
+        ipmiFeatures: this.loadIpmiFeatures().then(() => {
+          if (this.ipmi.isSerialOverLanSshKeySupported()) {
+            this.loadSshKey();
+          }
+
+          // load kvm features if IPMI is not activated
+          if (!this.ipmi.isActivated()) {
+            return this.$q
+              .all({
+                kvmFeatures: this.IpmiService.getKvmFeatures(
+                  this.serviceName,
+                ).catch((error) => {
+                  this.handleError(error, 'bm_server_kvm_error');
+                  return {};
+                }),
+                canOrderKvm: this.IpmiService.canOrderKvm(this.serviceName),
+              })
+              .then(({ kvmFeatures, canOrderKvm }) => {
+                this.kvm = new Kvm(kvmFeatures, canOrderKvm);
+              });
+          }
+          return true;
+        }),
+        taskInProgress: this.getTaskInProgress(),
+      })
+      .catch((error) => {
+        this.handleError(error, 'bm_server_ipmi_loading_error');
+        this.loader.error = true;
+      })
+      .finally(() => {
+        this.loader.loading = false;
+      });
+  }
+
+  $onDestroy() {
+    this.Polling.addKilledScope(this.$scope.$id);
+  }
+
+  loadIpmiFeatures() {
+    return this.IpmiService.getIpmiFeatures(this.serviceName).then(
+      (ipmiFeatures) => {
+        const urlkvm = `api/dedicated/server/${this.serviceName}/ipmi/connections/kvmipJnlp`;
+        this.ipmi = new Ipmi(
+          this.server,
+          ipmiFeatures,
+          this.$sce.trustAsResourceUrl(urlkvm),
+        );
+        return this.ipmi;
+      },
+    );
+  }
+
+  loadSshKey() {
+    return this.IpmiService.getSshKey()
+      .then((keys) => {
+        this.ssh.error = false;
+        this.ssh.list = keys;
+      })
+      .catch(() => {
+        this.ssh.error = true;
+      });
+  }
+
+  getIpmiNavigation() {
+    return this.IpmiService.ipmiGetConnection(
+      this.serviceName,
+      'serialOverLanURL',
+    )
+      .then((connect) => {
+        this.loader.navigationReady = connect.value;
+        this.$window.open(connect.value, '_blank', 'noopener');
+        this.handleSuccess('bm_server_ipmi_navigation_success');
+      })
+      .catch((error) => {
+        this.handleError(error, 'bm_server_ipmi_navigation_error');
+      })
+      .finally(() => {
+        this.loader.navigationLoading = false;
+        this.loader.buttonStart = false;
+      });
+  }
+
+  startIpmiTestStatus() {
+    this.disable.testActive = true;
+    this.disable.testIpmi = true;
+  }
+
+  startIpmiPollPing(task) {
+    this.startIpmiTestStatus();
+    this.setHttpState(false, true, false);
```

Comment:
i think having to call this function like that is kind of bad practice :
```js
this.setHttpState(false, true, false)
```

false, true, false doesn't means a lot when you are reading the code, it's kind of difficult to maintain

It looks like http state could simply be an enumeration, something like :
```js
this.httpState = this.STATE_LOADING;
this.httpState = this.STATE_ERROR;
// etc
```

if a state can only have one value at a time they is no need to have a separate variable for each state value

Reply:
oh i see that this code already exists so just ignore my comments regarding code review since it's not in the scope of your PR

Reply:
Its a valid comment. I am rewriting the IPMI feature as a reusable component that can be used in dedicated server and nutanix. Its better to remove the legacy code and write new code as per our latest guidelines. I will fix this comment.